tests/00_00_04-change-federation.js

```diff
@@ -0,0 +1,246 @@
+        const oldFederationAddress = commitFederationEvent.arguments.oldFederationBtcAddress;
+        expect(oldFederationAddress).to.be.equal(initialFederationAddress, 'The old federation address in the commit_federation event should now be the same as the initial.');
+
+        // Assert the new federation address in the commit federation event is the expected one.
+        const newActiveFederationAddress = commitFederationEvent.arguments.newFederationBtcAddress;
+        expect(newActiveFederationAddress).to.be.equal(expectedNewFederationAddress, 'The new federation address in the commit_federation event should be the expected one.');
+
+        // Assert the new federation btc public keys in the commit federation event are the expected ones.
+        const expectedNewFederationBtcPublicKeys = newFederationPublicKeys.map(federator => federator[KEY_TYPE_BTC]);
+        const newFederationBtcPublicKeysInString = commitFederationEvent.arguments.newFederationBtcPublicKeys;
+        const actualNewFederationBtcPublicKeys = getBtcPublicKeysInArray(newFederationBtcPublicKeysInString);
+        expect(actualNewFederationBtcPublicKeys).to.be.deep.equal(expectedNewFederationBtcPublicKeys, 'The new federation btc public keys should be the expected ones.');
+
+    });
+
+    it('should activate federation', async () => {
+
+        const federationActivationBlockNumber = commitFederationEvent.arguments.activationHeight;
+        const currentBlockNumber = await rskTxHelper.getClient().eth.getBlockNumber();
+        const blockDifference = federationActivationBlockNumber - currentBlockNumber;
+        // Mining enough blocks to activate the federation.
+        await rskUtils.mineAndSync(rskTxHelpers, blockDifference + 1);
+
+        // Assert the pending federation does not exist anymore.
+        const actualPendingFederationSize = Number(await bridge.methods.getPendingFederationSize().call());
+        const expectedFederationSize = -1;
+        expect(actualPendingFederationSize).to.be.equal(expectedFederationSize, 'The pending federation should not exist anymore.');
+
+        // Assert the active federation address is the expected one.
+        const newFederationAddress = await bridge.methods.getFederationAddress().call();
+        expect(newFederationAddress).to.not.be.equal(initialFederationAddress, 'The new active federation address should be different from the initial federation address.');
+        expect(newFederationAddress).to.be.equal(expectedNewFederationAddress, 'The new active federation address should be the expected one.');
+
+        // Assert the active federation redeem script is the expected ones.
+        const newActiveFederaetionErpRedeemScript = await bridge.methods.getActivePowpegRedeemScript().call();
+        expect(newActiveFederaetionErpRedeemScript).to.be.equal(expectedNewFederationErpRedeemScript, 'The new active federation erp redeem script should be the expected one.');
+
+    });
```

Comment:
Should we also advance until there is no longer a retiring federation? Just to be sure we are in a "normal" state when the tests begin

Reply:
Updated.